core/src/main/scala/cr/pulsar/Producer.scala

```diff
@@ -23,21 +23,23 @@ import cr.pulsar.internal.FutureLift._
 import fs2.concurrent.{ Topic => _ }
 import java.util.concurrent.TimeUnit
 
-import org.apache.pulsar.client.api.{ MessageId, ProducerBuilder, TypedMessageBuilder }
+import cr.pulsar.Producer.MessageKey
+import cr.pulsar.internal.TypedMessageBuilderOps._
+import org.apache.pulsar.client.api.{ MessageId, ProducerBuilder }
 
 import scala.concurrent.duration.FiniteDuration
 
 trait Producer[F[_], E] {
 
   /**
-    * It sends a message asynchronously.
+    * Sends a message asynchronously.
     */
-  def send(msg: E): F[MessageId]
+  def send(msg: E, key: MessageKey = MessageKey.Empty): F[MessageId]
```

Comment:
I'm not sure about this change, maybe it's better to introduce a separate method. Eg. `sendWithKey`?

Reply:
Why not an `Option` with default `None`?

Reply:
Yeah I'm in favor of adding that separate method.